RIOT-OS/RIOT
# tests/sys_crypto: add RFC3610 and NIST SP 800-38C tests

### Contribution description

Add the `RFC3610` and `NIST SP 800-38C` tests
I used the tests from `pycrypto` test suite as input to format the tests vectors. https://github.com/dlitz/pycrypto/blob/7acba5f3a6ff10f1424c309d0d34d2b713233019/lib/Crypto/SelfTest/Cipher/test_AES.py#L1489

I at first split `tests/unittests/tests-crypto` to its own test and refactored to prepare for adding the new tests.

I had to blacklist all `avr` and `msp430` boards as they fail with `left shift count >= width of type` and other bigger than 16bit issues.

### Testing procedure

The test should be run on boards to verify it is working for them.

For me the tests works locally on `native`, `samr21-xpro`, `iotlab-m3`:

```
make -C tests/sys_crypto/ flash test
...
main(): This is RIOT! (Version: 2019.10-devel-418-gc33e11-pr/tests/crypto/add_rfc_nist_tests)
..................................
OK (34 tests)
```

And it works on my local test server with:

    cc2650-launchpad frdm-k64f frdm-kw41z msba2 mulle nrf52dk nucleo-f103rb pba-d-01-kw2x sltb001a stm32f3discovery



#### No need to update `unittests`

I also tested that the boards in `BOARD_INSUFFICIENT_MEMORY` in `unittests` must still all be there (if they are not blacklisted as I do no know).


### Issues/PRs references

Split out of https://github.com/RIOT-OS/RIOT/pull/7968
Part of crypto/ccm: bugs in the implementation of CCM mode #8107

Now all `RFC3610` tests are working.
